drake/solvers/Optimization.h

```diff
@@ -331,70 +331,99 @@ class DRAKEOPTIMIZATION_EXPORT OptimizationProblem {
     return addQuadraticCost(Q, x_desired, variable_views);
   };
 
-  /** addConstraint
-   * @brief adds a constraint to the program.  method specializations ensure
-   * that the constraint gets added in the right way
+  /** addGenericConstraint
+   *
+   * @brief adds a generic constraint to the program.  This should
+   * only be used if a more specific type of constraint is not
+   * available, as it may require the use of a significantly more
+   * expensive solver.
    */
-  void addConstraint(std::shared_ptr<Constraint> con,
+  void addGenericConstraint(std::shared_ptr<Constraint> con,
                      VariableList const& vars) {
     problem_type.reset(problem_type->addGenericConstraint());
     generic_constraints.push_back(Binding<Constraint>(con, vars));
   }
-  void addConstraint(std::shared_ptr<Constraint> con) {
-    addConstraint(con, variable_views);
+  void addGenericConstraint(std::shared_ptr<Constraint> con) {
+    addGenericConstraint(con, variable_views);
   }
 
-  void addConstraint(std::shared_ptr<LinearEqualityConstraint> con,
-                     VariableList const& vars) {
-    problem_type.reset(problem_type->addLinearEqualityConstraint());
-    linear_equality_constraints.push_back(
-        Binding<LinearEqualityConstraint>(con, vars));
-  }
-
-  void addConstraint(std::shared_ptr<LinearEqualityConstraint> con) {
-    addConstraint(con, variable_views);
+  /** addLinearConstraint
```

Comment:
Is restating the function name a normal doxygen thing?  I don't have a problem with it per se, just seems a little redundant.